cmd/cluster/azure/create.go

```diff
@@ -265,6 +276,27 @@ func credentialSecret(namespace, name string) *corev1.Secret {
 }
 
 func (o *CreateOptions) GenerateNodePools(constructor core.DefaultNodePoolConstructor) []*hyperv1.NodePool {
+	var vmImage *hyperv1.AzureVMImage
+	if o.MarketplacePublisher == "" {
```

Comment:
can we please make sure this assumptions are communicated, enforced and unit tested cc @stevekuznetsov ptal